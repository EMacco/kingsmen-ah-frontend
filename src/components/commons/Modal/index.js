import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import './index.scss';

class Modal extends Component {
  constructor(props) {
    super(props);

    this.state = {

    };
  }

  render() {
    const {
      children, title, toggle, exitModal
    } = this.props;
    return (
      <div className={classNames('modal flex', { 'is-visible': toggle })}>
        <div className="modal-container text-center relative justify-center items-center">
          <button type="button" className="absolute modal-exit bg-white rounded-full py-1 px-3 shadow-lg cursor-pointer flex hover:bg-gray-300" onClick={exitModal}>
            <span className="icon ion-md-cancel">x</span>
          </button>
          <div className="modal_title">
            <h2 className="text-normal font-bold text-xl">{title}</h2>
          </div>
          <div className="modal_body">
            {children}
          </div>
        </div>
      </div>
    );
  }
}

Modal.propTypes = {
  toggle: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  exitModal: PropTypes.func.isRequired
};

Modal.defaultProp = {
  toggle: false,
  children: ''
};


export default Modal;
