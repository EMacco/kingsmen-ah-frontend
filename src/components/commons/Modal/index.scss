$modal-background-color: #fff;
$modal-bg: rgba(52, 54, 66, 0.9);

.modal {
  background: $modal-bg;
  cursor: pointer;
  height: 100%;
  left: 0;
  opacity: 0;
  overflow-y: auto;
  position: fixed;
  top: 0;
  transition: opacity 0.3s 0, visibility 0 0.3s;
  visibility: hidden;
  width: 100%;
  z-index: 21;

  &.is-visible {
    opacity: 1;
    transition: opacity 0.3s 0, visibility 0 0;
    visibility: visible;
  }

  &.modal-container {
    transform: translateY(0);
  }
}

.modal-container {
  background: $modal-background-color;
  border-radius: 0.25em;
  cursor: auto;
  margin: 3em auto 4em;
  padding: 30px;
  position: relative;
  width: 70%;
  z-index: 1000;
}

.modal-exit {
  right: -10px;
  top: -10px;
}
