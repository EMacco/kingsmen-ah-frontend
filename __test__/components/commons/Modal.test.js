import React from 'react';
import Modal from '@components/commons/Modal';


const props = {
    title: 'New Modal',
    toggle: false,
    exitModal: jest.fn()
};

const mockClick = jest.fn(props.toggle = false);

describe('<Modal />', () => {
    it('should render a modal correctly', () => {
        const shallowWrapper = shallow(<Modal {...props}><h1>Hello World</h1></Modal>);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Modal')).toBeTruthy();
    });

    it('should render a when toggle is true', () => {
      const shallowWrapper = shallow(<Modal {...props} toggle><h1>Hello World</h1></Modal>);

      expect(shallowWrapper.find('.modal-container')).toBeTruthy();
      expect(shallowWrapper.find('h2').text()).toEqual('New Modal');
    });

    it('should exit the toggle when exit is clicked', () => {
      const shallowWrapper = mount(<Modal {...props} exitModal={mockClick}><h1>Hello World</h1></Modal>);

      expect(shallowWrapper.props().toggle).toEqual(false);
    });
});
