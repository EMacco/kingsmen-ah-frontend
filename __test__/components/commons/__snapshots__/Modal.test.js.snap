// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render a modal correctly 1`] = `
<div
  className="modal flex"
>
  <div
    className="modal-container text-center relative justify-center items-center"
  >
    <button
      className="absolute modal-exit bg-white rounded-full py-1 px-3 shadow-lg cursor-pointer flex hover:bg-gray-300"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        className="icon ion-md-cancel"
      >
        x
      </span>
    </button>
    <div
      className="modal_title"
    >
      <h2
        className="text-normal font-bold text-xl"
      >
        New Modal
      </h2>
    </div>
    <div
      className="modal_body"
    >
      <h1>
        Hello World
      </h1>
    </div>
  </div>
</div>
`;
